proto/dataplane.proto

```diff
@@ -0,0 +1,214 @@
+  Error error = 1;
+  string message = 2;
+}
+
+enum Error {
+  ERROR_NONE = 0;
+  ERROR_VALIDATION_FAILED = 1;
+  ERROR_APPLY_FAILED = 2;
+  ERROR_UNKNOWN_ERROR = 3;
+}
+
+message GetConfigGenerationRequest {
+}
+
+message GetConfigGenerationResponse {
+  uint64 generation = 1;
+}
+
+/* ================================================ */
+/* ==== Interfaces (both underlay and overlay) ==== */
+/* ================================================ */
+
+message Interface {
+  string name = 1;
+  string ipaddr = 2;
+  IfType type = 3;
+  IfRole role = 4;
+  uint32 vlan = 5;
+  string macaddr = 6;
+  string portname = 7;
+}
+
+enum IfType {
+  IF_TYPE_ETHERNET = 0;
+  IF_TYPE_VLAN = 1;
+  IF_TYPE_LOOPBACK = 2;
+  IF_TYPE_VTEP = 3;
+}
+
+enum IfRole {
+  IF_ROLE_FABRIC = 0;
+  IF_ROLE_EXTERNAL = 1;
+}
+
+/* ================= */
+/* ==== Overlay ==== */
+/* ================= */
+
+message PeeringIPs {
+  oneof rule {
+    string cidr = 1;
+    string not = 2;
+  }
+}
+
+message PeeringAs {
+  oneof rule {
+    string cidr = 1;
+    string not = 2;
+  }
+}
+
+message Expose {
+  repeated PeeringIPs ips = 1;
+  repeated PeeringAs as = 2;
+}
+
+message PeeringEntryFor {
+  string vpc = 1;
+  repeated Expose expose = 2;
+}
+
+/* There can be only one peering per a pair of given VPCs */
+message VpcPeering {
+  string name = 1; /* unique: e.g. VPC-1--VPC-2 */
+  repeated PeeringEntryFor for = 2; /* it's guaranteed that only two entries are present */
+}
+
+message VPC {
+  string name = 1; /* unique (key) */
+  uint32 vni = 2;
+  repeated Interface interfaces = 3;
+}
+
+message Overlay {
+  repeated VPC vpcs = 1;
+  repeated VpcPeering peerings = 2;
+}
+
+/* ================== */
+/* ==== Underlay ==== */
+/* ================== */
+
+message BgpAddressFamilyIPv4 {
+  bool redistribute_connected = 1;
+  bool redistribute_static = 2;
+}
+
+message BgpAddressFamilyIPv6 {
+  bool redistribute_connected = 1;
+  bool redistribute_static = 2;
+}
+
+message BgpAddressFamilyL2vpnEvpn {
+  bool advertise_all_vni = 1;
+}
+
+enum BgpAF {
+  IPV4_UNICAST = 0;
+  IPV6_UNICAST = 1;
+  L2VPN_EVPN = 2;
+}
+
+message BgpNeighbor {
+  string address = 1;
+  string remote_asn = 2;
+  repeated BgpAF af_activate = 3;
+}
+
+message RouteMap {
+  string name = 1;
+  repeated string match_prefix_lists = 2;
+  string action = 3;
+  uint32 sequence = 4;
+}
+
+message RouterConfig {
+  string asn = 1;
+  string router_id = 2;
+  repeated BgpNeighbor neighbors = 3;
+  optional BgpAddressFamilyIPv4 ipv4_unicast = 4;
+  optional BgpAddressFamilyIPv6 ipv6_unicast = 5;
+  optional BgpAddressFamilyL2vpnEvpn l2vpn_evpn = 6;
+  repeated RouteMap route_maps = 7;
+}
+
+message VRF {
+  string name = 1;
+  repeated Interface interfaces = 2;
+  optional RouterConfig router = 3;
+}
+
+message Underlay {
+  repeated VRF vrf = 1;
+}
+
+/* ================ */
+/* ==== Device ==== */
+/* ================ */
+
+message Ports {
+  string name = 1;
+  string system_name = 2;
```

Comment:
If it has the same meaning as `Interface.portname`, let's please name it `portname` too

Reply:
it better to use system_name cause it looks confusing
`Port.portname` and `Port.name`